cardano-db-sync/src/Cardano/DbSync.hs

```diff
@@ -206,7 +204,7 @@ runDbSyncNodeNodeClient iomgr trce plugin nodeConfig (SocketPath socketPath) = d
   logInfo trce $ "localInitiatorNetworkApplication: connecting to node via " <> textShow socketPath
   networkState <- newNetworkMutableState
   txv <- newEmptyTMVarM @_ @(GenTx blk)
-  ncSubscriptionWorker_V1
+  ncSubscriptionWorker_V2
```

Comment:
@MarcFontaine could we use this pattern: https://github.com/input-output-hk/cardano-node/blob/master/cardano-cli/src/Cardano/CLI/Ops.hs#L436 and `nsSubscriptionWorker`; I'd like to deprecate `ncSubscriptionWorker_V1` and `ncSubscriptionWorker_V2` and remove them.

Reply:
Let me know when you fix that.

Reply:
I had looked it at when you suggested it on Slack, but had not been sure
if it is just a syntactic changes or if it breaks anything.
But  I changed it now.

Reply:
It does not break anything, and it's good to use the same pattern across whole ecosystem.

Reply:
`foldMapVersions` is like `foldMap` but for `Semigroup` rather than for `Monoid` (and specialised for `Versions` rather than for any `Semigroup`, just to make the api clearer). It folds the results with `(<>)` (one possible implementation is `foldMapVersions = \f -> foldr1 (<>) . fmap f`.  `ncSubscriptionWorker_V2` combines two versions explicitly with `(<>)` - so there's no difference.